Source/WebCore/platform/text/TextFlags.cpp

```diff
@@ -58,15 +60,67 @@ WTF::TextStream& operator<<(TextStream& ts, Kerning kerning)
     return ts;
 }
 
+template <typename T> using BinOp = std::function<T(T, T)>;
+
+template <typename T>
+T reduce(const std::vector<T>& vec, const BinOp<T>& binOp, std::optional<T> initial = { })
```

Comment:
Oh, I think I led you astray with a previous comment. I meant for this to be a member function of `WTF::Vector` (when I said `std` earlier that was an accidental typo). It should live right next to `WTF::Vector::map()`. Such a general function doesn't belong in TextFlags.cpp.

This function could also be accompanied by a few tests in TestWTF with a few test cases.

Please feel free to do this part in a separate patch.

Reply:
Yes, I was planning on moving this function to `WTF::Vector` that in a follow-up patch.